### ISSUE TITLE
More WP CLI capabilities

### ISSUE THREAD

**Issue by ianwijma** [mainwp/mainwp#96]
Hey MainWP,

I was curious if there is interest in more WP CLI capabilities. like adding sites from WP CLI. Because at work I already worked on a small add on plugin that allows such a functionality and found out how well written this plugin is.

So I was wondering if you guys are open for PR's that add WP CLI functionality to manage mainwp from CLI. And if so, which once you guys think are the most useful once? 

Also never contributes to OS software before really. So excuse me if I do something wrong.

**Comment by bogdan-mainwp** [comment]
@ianwijma Thanks for reaching out to us and showing the interest in our plugin. 

Yes, we are open for Pull Requests and we really appreciate that. Also, the additional WP-CLI functionality will be very useful to our users.

**Comment by ianwijma** [comment]
Awesome, So what do you guys think would be most interesting to implement into CLI first? I was thinking about more site management. list adding / removing / updating of sites.

**Comment by Xilonz** [comment]
I would love improvements to the CLI. Most important for me (to use with automated deployments) are:
wp mainwp site get --url="example.com"
wp mainwp site get --id="1"

wp mainwp site add 
wp mainwp site update

and for all wp-cli to have the --format option to display json output.

Now I have to use the wp eval function for this, and that isn't great.

**Comment by binaryfire** [comment]
+1 for something like wp mainwp site add

We automate the deployment of new client sites, and being able to use the CLI to fully automate adding child sites would be awesome

**Comment by planetahuevo** [comment]
For me adding a site would be the most important thing to do. Right now is a completely manual process.

**Comment by binaryfire** [comment]
@bogdan-mainwp Can we get any traction on this? I'm a pro license holder and this is something my agency desperately needs. We automate everything nowadays and adding / removing to MainWP is one of the only things we can't script.

**Comment by bogdan-mainwp** [comment]
Hi @binaryfire, at the moment, our dev team is pretty busy with projects that we have started and I don't have any kind of timeframe when we could dedicate some time for this. As I mentioned earlier, Pull Requests are more than welcome. 

For official feature requests, you can join the [MainWP Community](https://meta.mainwp.com) and post your suggestions to the Features section.

**Comment by dennisdornon** [comment]
More CLI commands have been added in version 4.1.3.  You can view them in the [KB](https://kb.mainwp.com/mainwp-wp-cli-commands/)

**Comment by planetahuevo** [comment]
@dennisdornon I cannot find any command to connect a site. 
without that, the whole wp cli is not that useful.
Do you want me to open another issue about that?

**Comment by Xilonz** [comment]
And what about output formatting? Is that supported? We mainly use wp cli for automation and computers can't read... easily.

**Comment by planetahuevo** [comment]
Yes. 
I was wondering what is the response to `wp mainwp site --site-update-plugins <id> ` if the id is empty or wrong one. 
How do we handle errors on our scripts if the docs do not provide any information on that?

**Comment by bogdan-mainwp** [comment]
Hi @planetahuevo, 

Regarding the command for adding a child site, we will look into this for the next update. Regarding responses, if you enter an invalid ID for commands where site ID is required, the following error message will be returned: **Site not found.**, if you forget to set ID, this: **Empty site id.**

Hi @Xilonz, 

Can you please provide me more details about your requirement? I am not sure that I am following.

**Comment by planetahuevo** [comment]
Thank you @bogdan-mainwp 
Does the old "mainwp plugin <id> --upgrade-all" commands still work or have they been deprecated? 
Thanks

**Comment by Xilonz** [comment]
@bogdan-mainwp if I'm not mistaken, the output of the cli command is not in a json format. Therefore it would not be usable for any CI/CD or integration  with other tools.

Example; if I want to do something with the site information, let's say the plugins installed on a site. The expected output of that command would be something like: `wp mainwp plugin --list --format=json`

```
{
installed: 15,
active: 5,
plugins: [{
  name: 'plugin-name'
  up-to-date: false
}]
}
```

**Comment by bogdan-mainwp** [comment]
@planetahuevo, yes, the old command is still available. 

@Xilonz, thanks for the details, I understand now. At the moment, the output format is not customizable, but I am adding this to our todo list. It will be added.

**Comment by planetahuevo** [comment]
@bogdan-mainwp can you add the documentation of the old commands to the same page?
I would also want to report an issue with that one. 
When id is missing on the old ones, it applies the command to all sites (it did a huge mess last week on my systems). That is the reason I asked with the new commands. :D

**Comment by bogdan-mainwp** [comment]
@planetahuevo, thanks for reporting it, and my apologies for the inconvenience. 

I will have our dev team review the old commands and make sure all work properly. Once that is done, we will add those to the documentation.

**Comment by planetahuevo** [comment]
Fantastic. 
No worries @bogdan-mainwp  I have put some checks in place myself so the variable is not empty anymore. :D
Thanks!

**Comment by bogdan-mainwp** [comment]
All further requests, please submit to the [MainWP Feedback](https://feedback.mainwp.com/).

**Comment by planetahuevo** [comment]
I do not see the point to be honest. 
I asked about the old commands in 2020 and years before that (via ticket). 
So I am sorry but I do not feel like investing more time reporting stuff to be ignored, and for sure I won´t be creating a new account on another website for that.
